### PR TITLE
Create recipe v1

### PR DESCRIPTION
**Pull request by prashanthich**
Screen Shot 2021-03-19 at 4 42 43 PM
Screen Shot 2021-03-19 at 5 11 53 PM

Build UI and DB
UI includes categories
How will categories be stored in the DB?
When you create a recipe the fields should be saved in the DB and you should be directed to the homepage

To Insert recipe categories
INSERT INTO `health_nutrition`.`recipe_category`(`id`,`description`,`name`)
VALUES (100,'Vegan','Vegan');
INSERT INTO `health_nutrition`.`recipe_category`(`id`,`description`,`name`)
VALUES (101,'Gluten free','Gluten free');
INSERT INTO `health_nutrition`.`recipe_category`(`id`,`description`,`name`)
VALUES (102,'Dairy free','Dairy free');
INSERT INTO `health_nutrition`.`recipe_category`(`id`,`description`,`name`)
VALUES (103,'Cultural','Cultural');
INSERT INTO `health_nutrition`.`recipe_category`(`id`,`description`,`name`)
VALUES (104,'Seasonal','Seasonal');

### CODE REVIEW

**Review by Rexzellius — COMMENTED** [review]
DB & UI seems to be functioning together. Well done.

**Review by revathivarini — APPROVED** [review]
Its great work!!

Thank you

File: src/main/java/org/launhcode/healthynutrition/controllers/RecipeController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import javax.validation.Valid;
+import java.util.Arrays;
+import java.util.List;
+
+@Controller
+@RequestMapping("recipe")
+public class RecipeController {
+    @Autowired
+    private RecipeRepository recipeRepository;
+    @Autowired
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/RecipeController.java:22]: minor: add a space between variables

File: src/main/java/org/launhcode/healthynutrition/controllers/RecipeController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private RecipeCategoryRepository recipeCategoryRepository;
+
+    @GetMapping("create")
+    public String displayCreateRecipeForm(Model model){
+        model.addAttribute("title", "Create Recipe");
+        model.addAttribute("categories", recipeCategoryRepository.findAll());
+        model.addAttribute(new Recipe());
+        return "recipe/create";
+    }
+    @PostMapping("create")
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/RecipeController.java:32]: minor: add a space between functions

File: src/main/java/org/launhcode/healthynutrition/models/Recipe.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.util.List;
+
+@Entity
+public class Recipe {
+    @Id
+    @GeneratedValue
+    private int id;
+    @NotBlank(message = "Name/Title is required")
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/models/Recipe.java:17]: minor: add spaces between variable names for readability

**Review by korina0430 — APPROVED** [review]
Very clean, concise PR! Great job!!

**Comment by korina0430** [comment]
i'm wondering if we can simplify the db's. Do we need one for category and categories?

**Pull request merged**